client/wayland/ibuswaylandim.c

```diff
@@ -681,6 +684,55 @@ _context_hide_preedit_text_cb (IBusInputContext *context,
     }
 }
 
+static void
+_context_delete_surrounding_text_cb(IBusInputContext *context,
+                                    gint              offset,
+                                    guint             nchars,
+                                    IBusWaylandIM    *wlim)
+{
+    IBusWaylandIMPrivate *priv;
+    const char *start, *end;
+    const char *before, *after;
+    const char *cursor;
+    uint32_t before_length;
+    uint32_t after_length;
+
+    g_return_if_fail (IBUS_IS_WAYLAND_IM (wlim));
+    priv = ibus_wayland_im_get_instance_private (wlim);
+
+    if (!priv->surrounding_text)
+        return;
+
+    offset = MIN (offset, 0);
+
+    start = priv->surrounding_text->text;
+    end = start + strlen (priv->surrounding_text->text);
+    cursor = start + priv->surrounding_text_cursor_pos;
+
+    before = g_utf8_offset_to_pointer (cursor, offset);
+    g_return_if_fail (before >= start);
+    after = g_utf8_offset_to_pointer (cursor, offset + nchars);
+    g_return_if_fail (after <= end);
+
+    before_length = cursor - before;
+    after_length = after - cursor;
+
+    switch (priv->version) {
+    case INPUT_METHOD_V1:
+        zwp_input_method_context_v1_delete_surrounding_text (priv->context,
+                                                    -before_length,
+                                                    before_length + after_length);
```

Comment:
Please put lines within 80 columns:

```suggestion
        zwp_input_method_context_v1_delete_surrounding_text (
                priv->context,
                -before_length,
                before_length + after_length);
```